### PR TITLE
Add PUT and PATCH operations for HubVirtualNetworkConnections resource

### PR DESCRIPTION
**Pull request by ginandag**
Added PUT and PATCH operations for HubVirtualNetworkConnections resource.

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/examples/HubVirtualNetworkConnectionUpdateTags.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                "key2": "value2"
+            }
+        }
+    },
+    "responses": {
+        "200": {
+            "body": [
+                {
+                    "name": "connection1",
+                    "id": "/subscriptions/subid/resourceGroups/rg1/providers/Microsofot.Network/virtualHubs/virtualHub1/virtualHubVnetConnections/connection1",
```
Comment (daschult) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/examples/HubVirtualNetworkConnectionUpdateTags.json]:
```suggestion
                    "id": "/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.Network/virtualHubs/virtualHub1/virtualHubVnetConnections/connection1",
```

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/examples/HubVirtualNetworkConnectionUpdateTags.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                    "etag": "w/\\00000000-0000-0000-0000-000000000000\\",
+                    "tags": {
+                        "key1": "value1",
+                        "key2": "value2"
+                    },
+                    "properties": {
+                        "provisioningState": "Succeeded",
+                        "remoteVirtualNetwork": {
+                            "id": "/subscriptions/subid/resourceGroups/rg1/providers/Microsofot.Network/virtualNetworks/vnet1"
```
Comment (daschult) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/examples/HubVirtualNetworkConnectionUpdateTags.json]:
```suggestion
                            "id": "/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.Network/virtualNetworks/vnet1"
```

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/examples/HubVirtualNetworkConnectionUpdateTags.json [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                        "allowRemoteVnetToUseHubVnetGateways": false
+                    }
+                }
+            ]
+        },
+        "201": {
+            "body": [
+                {
+                    "name": "connection1",
+                    "id": "/subscriptions/subid/resourceGroups/rg1/providers/Microsofot.Network/virtualHubs/virtualHub1/virtualHubVnetConnections/connection1",
```
Comment (daschult) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/examples/HubVirtualNetworkConnectionUpdateTags.json]:
```suggestion
                    "id": "/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.Network/virtualHubs/virtualHub1/virtualHubVnetConnections/connection1",
```

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/examples/HubVirtualNetworkConnectionUpdateTags.json [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+                    "etag": "w/\\00000000-0000-0000-0000-000000000000\\",
+                    "tags": {
+                        "key1": "value1",
+                        "key2": "value2"
+                    },
+                    "properties": {
+                        "provisioningState": "Succeeded",
+                        "remoteVirtualNetwork": {
+                            "id": "/subscriptions/subid/resourceGroups/rg1/providers/Microsofot.Network/virtualNetworks/vnet1"
```
Comment (daschult) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/examples/HubVirtualNetworkConnectionUpdateTags.json]:
```suggestion
                            "id": "/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.Network/virtualNetworks/vnet1"
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/virtualWan.json [hunk 5]
```diff
@@ -1090,6 +1090,145 @@
...
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The resource group name of the HubVirtualNetworkConnection."
+          },
```
Comment (daschult) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/virtualWan.json]: This `resourceGroupName` definition is used multiple times. I'd recommend extracting it to your definitions section, like you've done with `SubscriptionIdParameter`.

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/virtualWan.json [hunk 6]
```diff
@@ -1090,6 +1090,145 @@
...
+          {
+            "name": "virtualHubName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the parent Virtual Hub."
+          },
```
Comment (daschult) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/virtualWan.json]: This `virtualHubName` definition is used multiple times. I'd recommend extracting it to your definitions section, like you've done with `SubscriptionIdParameter`.

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/virtualWan.json [hunk 7]
```diff
@@ -1090,6 +1090,145 @@
...
+          {
+            "name": "connectionName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the HubVirtualNetworkConnection."
+          },
```
Comment (daschult) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/virtualWan.json]: This `connectionName` definition is used multiple times. I'd recommend extracting it to your definitions section, like you've done with `SubscriptionIdParameter`.

**Review by ghost — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/examples/HubVirtualNetworkConnectionUpdateTags.json (hunk 1 above)
Comment (ginandag) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/examples/HubVirtualNetworkConnectionUpdateTags.json]: Sorry for the typo. Fixed all occurrences.

**Comment by ravbhatnagar** [comment]
Signing off from ARM side.

**Review by ghost — APPROVED** [review]

**Review by MikhailTryakhov — CHANGES_REQUESTED** [review]
Only additive changes can be done in source code (examples are okay) for already published api-version=2018-12-01

**Comment by sergey-shandar** [comment]
@ginandag, @MikhailTryakhov what's your decision about this PR?

**Comment by ginandag** [comment]
> @ginandag, @MikhailTryakhov what's your decision about this PR?

We realized that some changes and bug fixes on the API side will not be pushed to production for a few weeks, so it isn’t a good idea to complete this PR now since we don’t want swagger changes out in production before API changes are available – even though the API operations are live, we need to fix some bugs. We don’t want people to create resources using the newly exposed ops in swagger since it might put the resources in a faulty state.

Is it possible to complete this PR later? I can also abandon this now, and create a new PR later… may be off of March branch. Let me know. Either way is fine with us.

**Comment by sergey-shandar** [comment]
@ginandag ok, closing for now. I may reference this PR to simplify next review.

**Comment by ginandag** [comment]
> @ginandag ok, closing for now. You may reference this PR to simplify next review.

Sure, I will start a new PR off of March/April (whichever is applicable) when we have API changes in PROD. Thanks.

**Pull request closed without merging**